### PR TITLE
Core: Wallet lookups do not iterate list

### PR DESCRIPTION
**Pull request by orogvany**
Resolve TODOs around wallet lookups.
Use LinkedHashMap to maintain order.

We're using a synchronized collection
yet usages look to all be protected with
synchronized.  It might be better to not use
synchronized wrapper, after double checking
synchronization.

Would like to discuss, and possibly add another
test before merging.

### CODE REVIEW

**Comment by orogvany** [comment]
Huh, I net removed code, but code coverage went down.  Very strange.

File: src/main/java/org/semux/core/Wallet.java [hunk 1]
```diff
@@ -314,7 +317,9 @@ public void setAccounts(List<Key> list) throws WalletLockedException {
     public Key getAccount(int idx) throws WalletLockedException {
         requireUnlocked();
 
-        return accounts.get(idx);
+        synchronized (accounts) {
+            return getAccounts().get(idx);
```
Comment (semuxdev) [inline src/main/java/org/semux/core/Wallet.java]: Just wanted to add a note here. One iteration over all keys are required here.

Comment (orogvany) [reply src/main/java/org/semux/core/Wallet.java]: Yes, it is a tradeoff.  As a list, that method is cheap, but the other 4-5 places are O(n) time.  With a map, the other methods are constant time, and this one is O(n) time.

As there's never likely to be over 100 accounts, it's all pretty trivial.  I could introduce another collection to keep track of index as well, but this seemed like an ok place to start.

**Comment by cryptokat** [comment]
related: #534

**Pull request merged**